contrib/bom.json.in

```diff
@@ -0,0 +1,45 @@
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.6",
+  "version": 1,
+  "components": [
+    {
+      "type": "application",
```

Comment:
I don't really know how this will be used for sure; but it might be a library in some contexts

Reply:
Right; I think the 'right' way to do this is to have multiple components inside the CycloneDX bom.json -- one with library, one with application. I'm not sure I'm too worried about exact semantics as we're only really targetting libraries used in the edk2 and coreboot build process

Reply:
If it's just the one field, I would say use meson configuration data to populate the string based on how it was built.  If it _needs_ to be hardcoded for how it's used; it's better to call it library.